### ISSUE TITLE
homectl create user fails (Transport endpoint is not conected)

### ISSUE THREAD

**Issue by Ginner** [picodotdev/alis#151]
I'm installing with the following relevant (imo) options:
```
LVM="true"
SYSTEMD_HOMED="true"
SYSTEMD_HOMED_STORAGE="luks"
BOOTLOADER="systemd"
```
Unfortunately it fails upon `homectl create user` with `Operation on home <user> failed: Transport endpoint is not connected` and `homectl activate <user>` with `Operation on home <user> failed: No home for user <user> known` and lastly `cp -a /home/ginner.home/. /mnt/home/ginner.home` fails with `No such file or directory`.

I think maybe you alludes to this in the code with: 
```
    ### something missing, inside alis this not works, after install the user is in state infixated
    ### after install and reboot this commands work
```
I tried rebooting and running the commands manually, without luck.

Any help is very appreciated! Thank you for your excellent work on this 'script' :)

**Comment by picodotdev** [comment]
Relevant info to review alis commands with the provided configuration.

https://wiki.archlinux.org/title/Systemd-homed#LUKS_home_directory
https://gist.github.com/dbehnke/ad19ca8f1ccf80aebca5

**Comment by Ginner** [comment]
Good info. Now I'm in no way strong in this, but at what point in the code, is the `IMAGE_PATH` being mounted? Should it be included in the `homectl create` with the `--image-path` as briefly described in the first link?

**Comment by picodotdev** [comment]
I think the problem is that systemd-homed for the luks storage expects the user home directory to be stored in "inside an encrypted LUKS volume inside a loopback file or any removable media".

For create the luks loopback file maybe the gist is usefull and maybe in --image-path/IMAGE_PATH must be specified that loopback file, but not sure also.

**Comment by Ginner** [comment]
Yeah, I think you're exactly right.

**Comment by picodotdev** [comment]
I think now should work as expected.

**Comment by Ginner** [comment]
Wow, You're fast :) 
I've tested it, but after the password prompt following `homectl create <user> --enforce-password-policy=no --timezone=<TZ> --language=<LANG> --storage=luks --image-path=/mnt/home/<USER>.home --fs-type=ext4 - G wheel,storage,optical` it fails with `Operation on home <USER> failed: Transport endpoint is not conected.` 

It seems to me, that `/mnt/home/<USER>.home` is not mounted.

**Comment by picodotdev** [comment]
Another try to solve Transport endpoint is not conected. Some magic sleep 5 inserted.

**Comment by picodotdev** [comment]
homectl only creates user homed image not mounts it.

I will try again, I do my testing in VirtualBox. Are you trying in a virtual machine?

**Comment by Ginner** [comment]
No, I'm testing on physical hardware. 

Yeah, `systemctl status systemd-homed.service` returns 'Active (running)'.

**Comment by picodotdev** [comment]
Tested again, no problem in VirtualBox. But I incremented the sleep time from 5 to 10.

Can you atach the full configuration file and describe your computer specs?

**Comment by Ginner** [comment]
Same result - `Operation on home <USER> failed: Transport endpoint is not conected`

Conf file:

[alis.conf.txt](https://github.com/picodotdev/alis/files/7756307/alis.conf.txt)

Computer specs:
ThinkPad X220i, Intel i5, 4 gb ram, ssd... Let me know if you need more than that :)

**Comment by picodotdev** [comment]
Tested in VirtualBox with exact same configuration you provided. There was a error in alis in the shell change to zsh for root user when using systemd-homed that I fixed, but that is a step lattter to your case.

I have no clues, need anyone, that what is happening in your ThinkPad. Maybe you can attach a screenshot or if you can test in VirtualBox or virtual machine.

**Comment by Ginner** [comment]
Thank you for all the time you're putting into this! Hmm.. It sounds weird - Unfortunately I'm not able to test in a VM rn. 

Heres a (rather shitty) picture of the error:
![alis-fail](https://user-images.githubusercontent.com/26798615/146984135-597658c2-8221-42f1-9502-2034cded8706.jpg)

**Comment by picodotdev** [comment]
Reproduced in VirtualBox! with your photo.

I think the problem is the password length. Use one longer than test3 like archlinux. I think the password must have a length > 6, dont ask me why is this but maybe there is a policy for this in luks or homectl. 

![transport endpont](https://user-images.githubusercontent.com/720376/146986578-bc0af3a4-fa64-43a5-8384-00c6d2d74da2.png)
o hint.

**Comment by Ginner** [comment]
Hmm.. okay. Though, I don't think password length is the issue, the one I took a picture of was the first install with a short password. Previously I've been using long (+12 character) passwords - The `testX` ones were just for the sake of the photo.

What did you change to reproduce?

**Comment by picodotdev** [comment]
I think use test3 as password for homectl and LUKS_PASSWORD, ROOT_PASSWORD, USER_PASSWORD was the only change that I did, the other case can be that I missed to type test3 on repeat the same.

**Comment by picodotdev** [comment]
I did one more test using test3 as password, you were right that was not the issue as my test finalized correctly. So, I really dont have clues what happened in the test that failed for me with the same error as you. Also, is strange that for me works and for you fails always in the same point with the same error.

The only suggestion for you is that you use other configuration if you dont have any new clue.

**Comment by Ginner** [comment]
Yeah, I'm not sure I'll go with homed anyways. Thank you for all your work on this, and thank you for helping me! I'll let you know if I figure it out.
